test/runtests.jl

```diff
@@ -101,15 +101,36 @@ test_tuples = [
 @testset "Shapefile" begin
 
 for test in test_tuples
-    shp = open(joinpath(@__DIR__, test.path)) do fd
-        read(fd, Shapefile.Handle)
+    for use_shx in (false, true)
+        shp = if use_shx
+            # this accesses .shp based on offsets in .shx
+            stempath, ext = splitext(test.path)
+            shxname = string(stempath, ".shx")
+            Shapefile.Handle(joinpath(@__DIR__, test.path), joinpath(@__DIR__, shxname))
+        else
+            # use .shp only
+            open(joinpath(@__DIR__, test.path)) do fd
+                read(fd, Shapefile.Handle)
+            end
+        end
+        shapes = unique(map(typeof, shp.shapes))
+        @test length(shapes) == 1
+        @test shapes[1] == test.geomtype
+      #  @test eltype(shp.shapes) == Union{test.geomtype,Missing}
```

Comment:
I don't know if these tests can be removed. Should the eltype be a `Union{xxx,Missing}`? Now it's not.

Also, I don't understand the purpose of the `MultiPatch` test.